### PR TITLE
Added gandi_livedns module

### PR DESCRIPTION
**Pull request by gthiemonge**
##### SUMMARY
Ansible module to control Gandi DNS service (https://www.gandi.net/en/domain)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
- gandi_livedns

### CODE REVIEW

File: lib/ansible/module_utils/gandi_livedns_api.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019 Gregory Thiemonge <gregory.thiemonge@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```
Comment (resmo) [inline lib/ansible/module_utils/gandi_livedns_api.py]: code in module_utils must not be GPL licensed.

File: lib/ansible/modules/net_tools/gandi_livedns.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        required_if=[
+            ('state', 'present', ['record', 'type', 'values']),
+            ('state', 'absent', ['record', 'type']),
+        ],
+    )
+
+    if not module.params['zone'] and not module.params['domain']:
+        module.fail_json(msg="At least one of zone and domain parameters need to be defined.")
+    if module.params['state'] == 'present' and module.params['ttl'] is None:
+        module.params['ttl'] = 10800
```
Comment (resmo) [inline lib/ansible/modules/net_tools/gandi_livedns.py:181]: is there a reason why you don't default this in the arg_spec?

**Review by resmo — CHANGES_REQUESTED** [review]
Well done, I only found an issue about licensing. But I have some questions, you put almost all logic into module_utils, usually module specific code not shared across other modules should go into the one module directly. Are there plans to use this in other gandi modules?

File: lib/ansible/module_utils/gandi_livedns_api.py (hunk 1 above)
Comment (gthiemonge) [reply lib/ansible/module_utils/gandi_livedns_api.py]: fixed in last push

File: lib/ansible/modules/net_tools/gandi_livedns.py (hunk 2 above)
Comment (gthiemonge) [reply lib/ansible/modules/net_tools/gandi_livedns.py:181]: 1. at least one of zone or domain is required, how could I define this in arg_spec?

2. ttl can be used as a filter when state is absent, and may be None if the user doesn't want to pass it explicitly

- {name: foo, type: A, state: absent} will remove foo A entry regardless the value of ttl
- {name: foo, type: A, ttl: 3600, state: absent} will remove foo A entry only if its ttl is 3600

So ttl is defaulted to something (10800) only if state is present.
Is there a way to have a default value in arg_spec only when state is present?

**Comment by gthiemonge** [comment]
> Are there plans to use this in other gandi modules?

I plan to send a PR for a gandi_livedns_facts module that makes use of this api.

**Comment by simonkrenger** [comment]
Any plans to merge this in the future? It just seems this does not pass the tests at the moment.

**Comment by sivel** [comment]
We are no longer accepting new modules or plugins in this repository. This pull request would need to be resubmitted to one of the new collection repositories, potentially https://github.com/ansible-collections/community.general

**Comment by simonkrenger** [comment]
@gthiemonge Any chance of having this new module submitted to the above ansible-collection?

**Comment by gthiemonge** [comment]
> @gthiemonge Any chance of having this new module submitted to the above ansible-collection?

Yes, I'll work on that. Thanks

**Comment by gthiemonge** [comment]
FYI PR to community.general repository proposed in https://github.com/ansible-collections/community.general/pull/328

**Comment by sivel** [comment]
Closing per above

**Pull request closed without merging**